services/frontend-service/pkg/handler/environments.go

```diff
@@ -17,28 +17,82 @@ Copyright 2023 freiheit.com*/
 package handler
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
+	"github.com/freiheit-com/kuberpult/pkg/api"
+	"golang.org/x/crypto/openpgp"
+	pgperrors "golang.org/x/crypto/openpgp/errors"
 	"net/http"
+)
 
-	xpath "github.com/freiheit-com/kuberpult/pkg/path"
+const (
+	// This maximum in-memory multipart size.
+	// It was chosen based on the assumption that we have < 10 envs with < 3MB manifests per env.
+	MAXIMUM_MULTIPART_SIZE = 32 * 1024 * 1024 // = 32Mi
 )
 
-func (s Server) HandleEnvironments(w http.ResponseWriter, req *http.Request, tail string) {
-	environment, tail := xpath.Shift(tail)
-	if environment == "" {
-		http.Error(w, "missing environment ID", http.StatusNotFound)
+func (s Server) handleCreateEnvironment(w http.ResponseWriter, req *http.Request, environment, tail string) {
+
+	if tail != "/" {
+		http.Error(w, fmt.Sprintf("Create Environment does not accept additional path arguments, got: '%s'", tail), http.StatusNotFound)
+		return
+	}
+	if err := req.ParseMultipartForm(MAXIMUM_MULTIPART_SIZE); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "Invalid body: %s", err)
+		return
+	}
+	if req.Body == nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "missing request body")
+		return
+	}
+
+	form := req.MultipartForm
+	envConfig := api.EnvironmentConfig{}
+
+	config, ok := form.Value["config"]
+	if ok {
+		err := json.Unmarshal([]byte(config[0]), &envConfig)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprintf(w, "Invalid body: %s", err)
+			return
+		}
+	} else {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Missing config in request body"))
 		return
 	}
 
-	function, tail := xpath.Shift(tail)
-	switch function {
-	case "applications":
-		s.handleApplications(w, req, environment, tail)
-	case "locks":
-		s.handleEnvironmentLocks(w, req, environment, tail)
-	case "releasetrain":
-		s.handleReleaseTrain(w, req, environment, tail)
-	default:
-		http.Error(w, fmt.Sprintf("unknown function '%s'", function), http.StatusNotFound)
+	if s.AzureAuth {
+		if signature, ok := form.Value["signature"]; ok {
+			if _, err := openpgp.CheckArmoredDetachedSignature(s.KeyRing, bytes.NewReader([]byte(config[0])), bytes.NewReader([]byte(signature[0]))); err != nil {
```

Comment:
```suggestion
	if signature, ok := form.Value["signature"]; ok {
        	if s.AzureAuth {
			if _, err := openpgp.CheckArmoredDetachedSignature(s.KeyRing, bytes.NewReader([]byte(config[0])), bytes.NewReader([]byte(signature[0]))); err != nil {
```

1) the signature should be validated, if it exists, even when azureAuth is false.
2) when azureAuth is true, there must be a signature
Your solution only does 2).

Reply:
Typo.
3. if signature exist and azureAuth=**false** -> validate